client/pool.go

```diff
@@ -123,18 +128,18 @@ func (p *Pool) connector(ctx context.Context, now time.Time) {
 		toCreate = p.client.PoolMaxSize - poolSize.Total
 	}
 
-	p.fillConnectionPool(ctx, now, toCreate)
+	return p.fillConnectionPool(ctx, now, toCreate)
 }
 
-func (p *Pool) fillConnectionPool(ctx context.Context, now time.Time, toCreate int) {
+func (p *Pool) fillConnectionPool(ctx context.Context, now time.Time, toCreate int) bool {
 	if p.client.RoundRobinConfig != nil {
 		if toCreate == 0 || len(p.connections) > 0 {
 			// Keep this up to date, or the logic will skip to the next server prematurely.
 			p.client.lastConn = now
 		} else if now.Sub(p.client.lastConn) > p.client.RetryInterval {
 			// We need more connections and the last successful connection was too long ago.
 			// Restart and skip to the next server in the round robin target list.
-			defer p.client.restart(ctx)
```

Comment:
bug: this got called twice. moving it up the stack and stopping the ticker at the same time should avoid a dup call.